### PR TITLE
(feat): run zq2 on containerized infrastructure

### PR DESCRIPTION
**Pull request by mauromedda**
the ZQ2 networks deployed on GCP will run on containers and it will enable also the support for scilla.
The IaC will now let us the possibility to deploy a set of core components (bootstrap+validators+apps) on a single region and optionally a set of decentralized validators in a multi-region fashion.
It also allows us to run multi-zonal deployment of ZQ2 nodes.
Removed the k8s dependencies. Now all the components are deployed on GCE VM by a provisioning script that selects which components install and run based on a given role value: validator or apps.

### CODE REVIEW

File: infra/tf/modules/node/main.tf [hunk 1]
```diff
@@ -168,6 +175,21 @@ sudo systemctl restart google-cloud-ops-agent
 apt update -y
 apt install -y pigz
 
+# Add Docker's official GPG key:
+apt install -y ca-certificates curl
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
```
Comment (rrw-zilliqa) [inline infra/tf/modules/node/main.tf:181]: You really want to put this in the repo, or anyone who compromises download.docker.com can compromise your image, which rather obviates the point of checking signatures.

File: infra/tf/main.tf [hunk 2]
```diff
@@ -229,6 +203,10 @@ data "google_compute_global_address" "api" {
   name = "api-${replace(var.subdomain, ".", "-")}"
 }
 
+data "google_compute_global_address" "bootstrap" {
+  name = "boostrap-${replace(var.subdomain, ".", "-")}"
```
Comment (rrw-zilliqa) [inline infra/tf/main.tf:207]: Did you want to call this bootstrap?

**Review by rrw-zilliqa — APPROVED** [review]
Looks good to me (modulo comments)

File: infra/tf/modules/node/main.tf (hunk 1 above)
Comment (mauromedda) [reply infra/tf/modules/node/main.tf:181]: Hey Richard, not really. I just did a proof of concept to see if the containerized version works. This has to be refactored properly Monday when I'm back. I just make this as draft pr to record what I'm doing

File: infra/tf/main.tf (hunk 2 above)
Comment (mauromedda) [reply infra/tf/main.tf:207]: As per above, I kept the current naming to do a test on the containered version. Bootstrap is just how has been named this node currently

Comment (JamesHinshelwood) [reply infra/tf/main.tf:205]: @rrw-zilliqa What made you think that bootstrap was the incorrect name? I'm not against renaming, but in my mind bootstrap was an accurate name. So I'm worried that we have a different understanding of what this node does.

I thought bootstrap was a good name to describe the node which new nodes contact in order to learn about the existence of the rest of the network.

Comment (rrw-zilliqa) [reply infra/tf/main.tf:205]: bootstrap is fine. "boostrap" seems odd :-p

**Review by pavlops — APPROVED** [review]

**Review by pavlops — APPROVED** [review]

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**